### ISSUE TITLE
Error in precompilation on v0.7.0-alpha on mac

### ISSUE THREAD

**Issue by GarrettJenkinson** [JuliaIO/JLD.jl#211]
I have hdf5 installed via homebrew outside of julia, and the JLD package installs and loads correctly on my v0.6.2, but errors during precompilation on the v0.7.0-alpha binary I downloaded:

```
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.6.2 (2017-12-13 18:08 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-apple-darwin14.5.0

julia> using JLD

julia> versioninfo()
Julia Version 0.6.2
Commit d386e40c17 (2017-12-13 18:08 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin14.5.0)
  CPU: Intel(R) Core(TM) i7-3615QM CPU @ 2.30GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, ivybridge)


               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.7.0-alpha.0 (2018-05-31 00:07 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-apple-darwin14.5.0

julia> using JLD
[ Info: Precompiling module JLD
┌ Warning: The function `cfunction` is now written as a macro `@cfunction`.
│   caller = register_blosc() at blosc_filter.jl:122
└ @ HDF5 blosc_filter.jl:122
┌ Warning: The function `cfunction` is now written as a macro `@cfunction`.
│   caller = register_blosc() at blosc_filter.jl:123
└ @ HDF5 blosc_filter.jl:123
WARNING: importing deprecated binding Base.ObjectIdDict into JLD.
WARNING: Base.ObjectIdDict is deprecated, use IdDict{Any, Any} instead.
  likely near /Users/gar/.julia/packages/JLD/D7dp/src/JLD.jl:41
ERROR: LoadError: LoadError: UndefVarError: SimpleVector not defined
Stacktrace:
 [1] top-level scope
 [2] include at ./boot.jl:314 [inlined]
 [3] include_relative(::Module, ::String) at ./loading.jl:1071
 [4] include(::Module, ::String) at ./sysimg.jl:29
 [5] include(::String) at /Users/gar/.julia/packages/JLD/D7dp/src/JLD.jl:3
 [6] top-level scope
 [7] include at ./boot.jl:314 [inlined]
 [8] include_relative(::Module, ::String) at ./loading.jl:1071
 [9] include(::Module, ::String) at ./sysimg.jl:29
 [10] top-level scope
 [11] eval at ./boot.jl:316 [inlined]
 [12] eval(::Expr) at ./client.jl:394
 [13] macro expansion at ./none:3 [inlined]
 [14] top-level scope at ./<missing>:0
in expression starting at /Users/gar/.julia/packages/JLD/D7dp/src/jld_types.jl:15
in expression starting at /Users/gar/.julia/packages/JLD/D7dp/src/JLD.jl:110
ERROR: Failed to precompile JLD to /Users/gar/.julia/compiled/v0.7/JLD/l8ij.ji.
Stacktrace:
 [1] error at ./error.jl:33 [inlined]
 [2] compilecache(::Base.PkgId) at ./loading.jl:1207
 [3] _require(::Base.PkgId) at ./loading.jl:1007
 [4] require(::Base.PkgId) at ./loading.jl:878
 [5] require(::Module, ::Symbol) at ./loading.jl:873

julia> versioninfo()
Julia Version 0.7.0-alpha.0
Commit 22590d529d (2018-05-31 00:07 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin14.5.0)
  CPU: Intel(R) Core(TM) i7-3615QM CPU @ 2.30GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, ivybridge)
Environment:
  JULIA_NUM_THREADS = 4

```

I understand that 0.7 is in alpha, so feel free to close this issue if I should wait until a beta or later release. Also let me know if there is any further information I can provide to be helpful here.

Also FWIW, JLD2 is also failing to precompile on 0.7-alpha on this machine:

```
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.7.0-alpha.0 (2018-05-31 00:07 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-apple-darwin14.5.0

julia> using Pkg

julia> Pkg.add("JLD2")
  Updating registry at `~/.julia/registries/Uncurated`
  Updating git-repo `https://github.com/JuliaRegistries/Uncurated.git`
 Resolving package versions...
Downloaded JLD2 ─ v0.0.5
  Updating `~/.julia/environments/v0.7/Project.toml`
  [033835bb] + JLD2 v0.0.5
  Updating `~/.julia/environments/v0.7/Manifest.toml`
  [033835bb] + JLD2 v0.0.5

julia> using JLD2
[ Info: Precompiling module JLD2
WARNING: importing deprecated binding Base.Void into JLD2.
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near /Users/gar/.julia/packages/JLD2/6s4K/src/mmapio.jl:19
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near /Users/gar/.julia/packages/JLD2/6s4K/src/mmapio.jl:19
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near /Users/gar/.julia/packages/JLD2/6s4K/src/mmapio.jl:19
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near /Users/gar/.julia/packages/JLD2/6s4K/src/mmapio.jl:19
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near /Users/gar/.julia/packages/JLD2/6s4K/src/mmapio.jl:19
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near /Users/gar/.julia/packages/JLD2/6s4K/src/mmapio.jl:19
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near /Users/gar/.julia/packages/JLD2/6s4K/src/mmapio.jl:32
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near /Users/gar/.julia/packages/JLD2/6s4K/src/mmapio.jl:32
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near /Users/gar/.julia/packages/JLD2/6s4K/src/mmapio.jl:42
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near /Users/gar/.julia/packages/JLD2/6s4K/src/mmapio.jl:46
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near /Users/gar/.julia/packages/JLD2/6s4K/src/mmapio.jl:132
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near /Users/gar/.julia/packages/JLD2/6s4K/src/mmapio.jl:254
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near /Users/gar/.julia/packages/JLD2/6s4K/src/mmapio.jl:254
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near /Users/gar/.julia/packages/JLD2/6s4K/src/mmapio.jl:254
┌ Warning: `isbits(#= deprecated.jl:1678 =# @nospecialize(t::Type))` is deprecated, use `isbitstype(t)` instead.
│   caller = define_packed(::DataType) at misc.jl:7
└ @ JLD2 misc.jl:7
┌ Warning: `unshift!` is deprecated, use `pushfirst!` instead.
│   caller = define_packed(::DataType) at misc.jl:10
└ @ JLD2 misc.jl:10
WARNING: importing deprecated binding Base.ObjectIdDict into JLD2.
WARNING: Base.ObjectIdDict is deprecated, use IdDict{Any, Any} instead.
  likely near /Users/gar/.julia/packages/JLD2/6s4K/src/JLD2.jl:135
WARNING: Base.ObjectIdDict is deprecated, use IdDict{Any, Any} instead.
  likely near /Users/gar/.julia/packages/JLD2/6s4K/src/JLD2.jl:135
┌ Warning: `isbits(#= deprecated.jl:1678 =# @nospecialize(t::Type))` is deprecated, use `isbitstype(t)` instead.
│   caller = define_packed(::DataType) at misc.jl:7
└ @ JLD2 misc.jl:7
┌ Warning: `unshift!` is deprecated, use `pushfirst!` instead.
│   caller = define_packed(::DataType) at misc.jl:10
└ @ JLD2 misc.jl:10
┌ Warning: `isbits(#= deprecated.jl:1678 =# @nospecialize(t::Type))` is deprecated, use `isbitstype(t)` instead.
│   caller = define_packed(::DataType) at misc.jl:7
└ @ JLD2 misc.jl:7
┌ Warning: `unshift!` is deprecated, use `pushfirst!` instead.
│   caller = define_packed(::DataType) at misc.jl:10
└ @ JLD2 misc.jl:10
┌ Warning: `isbits(#= deprecated.jl:1678 =# @nospecialize(t::Type))` is deprecated, use `isbitstype(t)` instead.
│   caller = define_packed(::DataType) at misc.jl:7
└ @ JLD2 misc.jl:7
┌ Warning: `unshift!` is deprecated, use `pushfirst!` instead.
│   caller = define_packed(::DataType) at misc.jl:10
└ @ JLD2 misc.jl:10
┌ Warning: `isbits(#= deprecated.jl:1678 =# @nospecialize(t::Type))` is deprecated, use `isbitstype(t)` instead.
│   caller = define_packed(::DataType) at misc.jl:7
└ @ JLD2 misc.jl:7
┌ Warning: `unshift!` is deprecated, use `pushfirst!` instead.
│   caller = define_packed(::DataType) at misc.jl:10
└ @ JLD2 misc.jl:10
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near /Users/gar/.julia/packages/JLD2/6s4K/src/dataspaces.jl:43
ERROR: LoadError: LoadError: syntax: local variable name "x" conflicts with an argument
Stacktrace:
 [1] include at ./boot.jl:314 [inlined]
 [2] include_relative(::Module, ::String) at ./loading.jl:1071
 [3] include(::Module, ::String) at ./sysimg.jl:29
 [4] include(::String) at /Users/gar/.julia/packages/JLD2/6s4K/src/JLD2.jl:3
 [5] top-level scope
 [6] include at ./boot.jl:314 [inlined]
 [7] include_relative(::Module, ::String) at ./loading.jl:1071
 [8] include(::Module, ::String) at ./sysimg.jl:29
 [9] top-level scope
 [10] eval at ./boot.jl:316 [inlined]
 [11] eval(::Expr) at ./client.jl:394
 [12] macro expansion at ./none:3 [inlined]
 [13] top-level scope at ./<missing>:0
in expression starting at /Users/gar/.julia/packages/JLD2/6s4K/src/dataspaces.jl:47
in expression starting at /Users/gar/.julia/packages/JLD2/6s4K/src/JLD2.jl:426
WARNING: Base.Void is deprecated, use Nothing instead.
 in module JLD2
WARNING: Base.Void is deprecated, use Nothing instead.
 in module JLD2
WARNING: Base.Void is deprecated, use Nothing instead.
 in module JLD2
WARNING: Base.Void is deprecated, use Nothing instead.
 in module JLD2
WARNING: Base.Void is deprecated, use Nothing instead.
 in module JLD2
WARNING: Base.Void is deprecated, use Nothing instead.
 in module JLD2
WARNING: Base.Void is deprecated, use Nothing instead.
 in module JLD2
WARNING: Base.Void is deprecated, use Nothing instead.
 in module JLD2
ERROR: Failed to precompile JLD2 to /Users/gar/.julia/compiled/v0.7/JLD2/O1Ey.ji.
Stacktrace:
 [1] error at ./error.jl:33 [inlined]
 [2] compilecache(::Base.PkgId) at ./loading.jl:1207
 [3] _require(::Base.PkgId) at ./loading.jl:1007
 [4] require(::Base.PkgId) at ./loading.jl:878
 [5] require(::Module, ::Symbol) at ./loading.jl:873
```

But it looks like this is [a known issue in JLD2](https://github.com/simonster/JLD2.jl/issues/61). I did not see a similiar issue in this JLD repository.

**Comment by garrison** [comment]
Looks like [the fix there](https://github.com/simonster/JLD2.jl/pull/66) was pretty simple.

**Comment by JeffBezanson** [comment]
Fixed by #227.
